src/components/list/moto_list.tsx

```diff
@@ -65,37 +103,56 @@ export default function MotorcycleList({ motorcycles, setMotorcycles }: Motorcyc
       <Dialog open={editIndex !== null} onClose={handleEditClose}>
         <DialogTitle>Edit Motorcycle</DialogTitle>
         <DialogContent>
-          <Stack direction="column" spacing={2} padding={2}> 
+          <Stack direction="column" spacing={2} padding={2}>
             <FormControl sx={{ width: 300 }}>
```

Comment:
Here you should use a `handleSubmit` from a react-hook-form. It will automatically validate the form before submit and throw errors if any exists. Check the doc https://react-hook-form.com/docs/useform/handlesubmit
Keep mind, that you also set a prop `component="form"` to your container